nomad/job_endpoint_validators.go

```diff
@@ -47,9 +47,59 @@ func (c jobNamespaceConstraintCheckHook) Validate(job *structs.Job) (warnings []
 			)
 		}
 	}
+
+	var disallowedNetworkModes []string
+	for _, tg := range job.TaskGroups {
+		for _, network := range tg.Networks {
+			if !taskValidateNetworkMode(network, ns) {
+				network_mode := "host"
+				if len(network.Mode) > 0 {
+					network_mode = network.Mode
+				}
```

Comment:
We end up having to repeat this logic in the function. Maybe change the signature of `taskValidateNetworkMode` so that it returns the mode name it checked?